lib/buttons-and-links/KButtonGroup.vue

```diff
@@ -19,6 +9,16 @@
 
   export default {
     name: 'KButtonGroup',
+    render(createElement) {
+      var children = [];
+      this.$slots.default.forEach(element => {
+        if (element.tag) {
+          children.push(createElement('span', { class: 'button-group-item' }, [element]));
+        }
+      });
+
+      return createElement('div', { class: 'button-group' }, children);
+    },
```

Comment:
Does doing this (as compared to the `div` in the template) make it more dynamic?

Reply:
Using a slot in a template, all items that get passed in are wrapped by whatever is around the template, for example:

```html
  <div class="button-group">
    <button />
    <button />
    <!-- ... -->
    <button />
  </div>
```

I wasn't able to figure out a way to increase the vertical margins of the buttons while still keeping them as `inline-block` so they'd wrap to the next line without putting each one in a wrapper node.

Using the render function one can iterate over the elements passed into the slot. Here I used that to wrap each one in an inline-block span with a fixed larger height like this:

```html
  <div class="button-group">
    <span class="button-group-item"><button /></span>
    <span class="button-group-item"><button /></span>
    <!-- ... -->
    <span class="button-group-item"><button /></span>
  </div>
```